File: Objective-C Sample Code/Chapter 4/Going Further/Top10/Top10 WatchKit App Extension/InterfaceController.m
//
//  InterfaceController.m
//  Top10 WatchKit App Extension
//
//  Created by Cory Bohon on 12/5/16.
//  Copyright © 2016 Cory Bohon. All rights reserved.
//

#import "InterfaceController.h"
#import "NetworkController.h"
#import "JSONItem.h"
#import "MainRow.h"

@interface InterfaceController()

@property (nonatomic, weak) IBOutlet WKInterfaceTable *table;
@property (nonatomic, strong) NSArray *jsonItems;

@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context
{
    [super awakeWithContext:context];
    [self reloadData];
}

- (void)reloadData
{
    [[NetworkController sharedNetworkController] retrieveJSONFeedWithCompletionHandler:^(NSError *error, NSArray *objects) {
        
        [self.table setNumberOfRows:objects.count withRowType:@"MainRow"];
        
        for (NSUInteger row = 0; row < self.table.numberOfRows; row++) {
            MainRow *currentRow = [self.table rowControllerAtIndex:row];
            [currentRow configureWithJSONItem:[objects objectAtIndex:row] atIndex:row];
        }
        
        self.jsonItems = objects;
        
    } forNumberOfItems:10];
}

- (IBAction)reloadMenuButtonTapped:(id)sender
{
    [self reloadData];
}

@end

File: Swift Sample Code/Chapter 4/Going Further/Top10/Top 10 WatchKit App Extension/InterfaceController.swift
//
//  InterfaceController.swift
//  Top 10 WatchKit App Extension
//
//  Created by Cory Bohon on 2/13/17.
//  Copyright © 2017 Cory Bohon. All rights reserved.
//

import WatchKit
import Foundation


class InterfaceController: WKInterfaceController {

    @IBOutlet weak var table: WKInterfaceTable?
    
    var jsonItems: Array<Any>?
    
    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        self.setTitle("Top 10")
        
        self.reloadData()
    }
    
    @IBAction func reloadData() {
        NetworkController.shared().retrieveJSONFeed(completionHandler: { (error, objects) in
            
            self.table?.setNumberOfRows(10, withRowType: "MainRow")
            
            for row in 0..<10 {
                let currentRow: MainRow = self.table?.rowController(at: row) as! MainRow
                currentRow.configure(item: objects[row] as! JSONItem, index: row)
            }
            
            self.jsonItems = objects
            
        }, forNumberOfItems: 10)
    }
}
